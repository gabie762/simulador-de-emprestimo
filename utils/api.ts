// utils/api.ts
import { Produto, Resultado } from '../types';

// Configuração da API
const API_BASE_URL = 'https://sua-api.com/api'; // Substitua pela URL da sua API

// Tipos para requests/responses
export interface CadastrarProdutoRequest {
  nome: string;
  taxaJurosAnual: number;
  prazoMaximo: number;
}

export interface SimulacaoRequest {
  produtoId: string;
  valor: number;
  prazo: number;
}

export interface SimulacaoResponse {
  produto: Produto;
  valorEmprestimo: number;
  prazo: number;
  taxaJurosAnual: number;
  taxaJurosMensal: number;
  valorParcela: number;
  totalPagar: number;
  totalJuros: number;
  memoriaCalculo?: {
    mes: number;
    parcela: number;
    juros: number;
    amortizacao: number;
    saldoDevedor: number;
  }[];
}

// Função helper para fazer requests
async function apiRequest<T>(
  endpoint: string,
  options: RequestInit = {}
): Promise<T> {
  const url = `${API_BASE_URL}${endpoint}`;
  
  const defaultHeaders = {
    'Content-Type': 'application/json',
    'Accept': 'application/json',
    // Adicione headers de autenticação se necessário
    // 'Authorization': `Bearer ${token}`,
  };

  const config: RequestInit = {
    ...options,
    headers: {
      ...defaultHeaders,
      ...options.headers,
    },
  };

  try {
    const response = await fetch(url, config);
    
    if (!response.ok) {
      throw new Error(`HTTP ${response.status}: ${response.statusText}`);
    }
    
    const data = await response.json();
    return data;
  } catch (error) {
    console.error(`Erro na API (${endpoint}):`, error);
    throw error;
  }
}

// GET /produtos – Listar produtos
export async function getProdutos(): Promise<Produto[]> {
  try {
    const produtos = await apiRequest<Produto[]>('/produtos', {
      method: 'GET',
    });
    return produtos;
  } catch (error) {
    console.error('Erro ao buscar produtos:', error);
    throw new Error('Não foi possível carregar os produtos do servidor.');
  }
}

// POST /produtos – Cadastrar produto
export async function postProduto(produto: CadastrarProdutoRequest): Promise<Produto> {
  try {
    const novoProduto = await apiRequest<Produto>('/produtos', {
      method: 'POST',
      body: JSON.stringify(produto),
    });
    return novoProduto;
  } catch (error) {
    console.error('Erro ao cadastrar produto:', error);
    
    // Tratamento de erros específicos
    if (error instanceof Error) {
      if (error.message.includes('409')) {
        throw new Error('PRODUTO_DUPLICADO');
      }
      if (error.message.includes('400')) {
        throw new Error('Dados do produto inválidos.');
      }
    }
    
    throw new Error('Não foi possível cadastrar o produto.');
  }
}

// POST /simulações – Realizar simulação de empréstimo
export async function postSimulacao(simulacao: SimulacaoRequest): Promise<Resultado> {
  try {
    const resultado = await apiRequest<SimulacaoResponse>('/simulacoes', {
      method: 'POST',
      body: JSON.stringify(simulacao),
    });
    
    // Converte o response da API para o formato esperado pelo app
    const resultadoFormatado: Resultado = {
      produto: resultado.produto.nome,
      valorEmprestimo: resultado.valorEmprestimo.toFixed(2),
      prazo: resultado.prazo,
      taxaJurosAnual: resultado.taxaJurosAnual.toFixed(2) + '%',
      taxaJurosMensal: resultado.taxaJurosMensal.toFixed(2) + '%',
      valorParcela: resultado.valorParcela.toFixed(2),
      totalPagar: resultado.totalPagar.toFixed(2),
      totalJuros: resultado.totalJuros.toFixed(2),
      memoriaCalculo: resultado.memoriaCalculo?.map(item => ({
        mes: item.mes,
        parcela: item.parcela.toFixed(2),
        juros: item.juros.toFixed(2),
        amortizacao: item.amortizacao.toFixed(2),
        saldoDevedor: item.saldoDevedor.toFixed(2),
      })) || [],
    };
    
    return resultadoFormatado;
  } catch (error) {
    console.error('Erro ao realizar simulação:', error);
    
    // Tratamento de erros específicos
    if (error instanceof Error) {
      if (error.message.includes('400')) {
        throw new Error('Dados da simulação inválidos.');
      }
      if (error.message.includes('404')) {
        throw new Error('Produto não encontrado.');
      }
    }
    
    throw new Error('Não foi possível realizar a simulação.');
  }
}

// Função para deletar produto (se a API suportar)
export async function deleteProduto(id: string): Promise<boolean> {
  try {
    await apiRequest(`/produtos/${id}`, {
      method: 'DELETE',
    });
    return true;
  } catch (error) {
    console.error('Erro ao deletar produto:', error);
    throw new Error('Não foi possível deletar o produto.');
  }
}

// Função para testar conectividade com a API
export async function testApiConnection(): Promise<boolean> {
  try {
    // Tenta fazer uma requisição simples para verificar se a API está acessível
    await apiRequest('/produtos', { method: 'GET' });
    return true;
  } catch (error) {
    console.error('API não está acessível:', error);
    return false;
  }
}
