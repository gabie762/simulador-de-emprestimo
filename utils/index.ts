import { Produto, Resultado } from '../types';
import { Platform } from 'react-native';

export const mockProdutos: Produto[] = [
  { id: '1', nome: 'Crédito Pessoal Padrão', taxaJurosAnual: 25.0, prazoMaximo: 48 },
  { id: '2', nome: 'Crédito Imobiliário', taxaJurosAnual: 9.5, prazoMaximo: 360 },
  { id: '3', nome: 'Financiamento Veicular', taxaJurosAnual: 18.2, prazoMaximo: 60 },
];

// Helper para normalizar estilos shadow para web
export const normalizeShadow = (shadowStyle: any) => {
  if (Platform.OS === 'web') {
    const { shadowColor, shadowOffset, shadowOpacity, shadowRadius, ...rest } = shadowStyle;
    
    if (shadowColor && shadowOffset && shadowOpacity !== undefined && shadowRadius) {
      return {
        ...rest,
        boxShadow: `${shadowOffset.width}px ${shadowOffset.height}px ${shadowRadius}px rgba(0, 0, 0, ${shadowOpacity})`,
      };
    }
  }
  
  return shadowStyle;
};

export const calcularSimulacao = (
  valorEmprestimo: number,
  prazo: number,
  produto: Produto
): Resultado => {
  const taxaMensal = (produto.taxaJurosAnual / 12) / 100;
  const parcela = valorEmprestimo * (taxaMensal * Math.pow(1 + taxaMensal, prazo)) / (Math.pow(1 + taxaMensal, prazo) - 1);
  const total = parcela * prazo;
  const totalJuros = total - valorEmprestimo;
  
  let saldoDevedor = valorEmprestimo;
  const memoriaCalculo = [];
  
  for (let i = 1; i <= prazo; i++) {
    const jurosMes = saldoDevedor * taxaMensal;
    const amortizacao = parcela - jurosMes;
    saldoDevedor -= amortizacao;
    memoriaCalculo.push({ 
      mes: i, 
      parcela: parcela.toFixed(2), 
      juros: jurosMes.toFixed(2), 
      amortizacao: amortizacao.toFixed(2), 
      saldoDevedor: Math.abs(saldoDevedor).toFixed(2) 
    });
  }

  return {
    produto: produto.nome,
    valorEmprestimo: valorEmprestimo.toFixed(2),
    prazo: prazo,
    taxaJurosAnual: produto.taxaJurosAnual.toFixed(2),
    taxaJurosMensal: (taxaMensal * 100).toFixed(4),
    valorParcela: parcela.toFixed(2),
    totalPagar: total.toFixed(2),
    totalJuros: totalJuros.toFixed(2),
    memoriaCalculo,
  };
};
