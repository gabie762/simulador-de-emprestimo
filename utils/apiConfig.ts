// utils/apiConfig.ts
import * as ApiReal from './api';
import * as ApiMock from './apiMock';

// Configuração para alternar entre Mock e API real
export const USE_REAL_API = false; // Mude para true quando quiser usar a API real

// Exporta as funções baseado na configuração
export const getProdutos = USE_REAL_API ? ApiReal.getProdutos : ApiMock.getProdutos;
export const postProduto = USE_REAL_API ? ApiReal.postProduto : ApiMock.postProduto;
export const deleteProduto = USE_REAL_API ? ApiReal.deleteProduto : ApiMock.deleteProduto;

// Função de simulação (só existe na API real, no mock usamos o cálculo local)
export const postSimulacao = USE_REAL_API ? ApiReal.postSimulacao : null;

// Função para testar conectividade (só para API real)
export const testApiConnection = USE_REAL_API ? ApiReal.testApiConnection : null;
