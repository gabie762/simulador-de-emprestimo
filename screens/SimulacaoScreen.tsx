import React, { useState } from 'react';
import { ActivityIndicator, Animated, ScrollView, StyleSheet, Text, TextInput, TouchableOpacity } from 'react-native';
import { ProdutoCompactSelector, ProdutoSelectorModal } from '../components';
import { globalStyles } from '../styles';
import { Produto, SimulacaoScreenProps } from '../types';
import { calcularSimulacao } from '../utils';
import { postSimulacao } from '../utils/apiConfig';

const SimulacaoScreen: React.FC<SimulacaoScreenProps> = ({ 
  produtos, 
  goHome, 
  setResultado, 
  setModalVisible 
}) => {
  const [valor, setValor] = useState('');
  const [prazo, setPrazo] = useState('');
  const [produtoSelecionado, setProdutoSelecionado] = useState<Produto | null>(
    produtos.length > 0 ? produtos[0] : null
  );
  const [isSimulating, setIsSimulating] = useState(false);
  const [modalProdutoVisible, setModalProdutoVisible] = useState(false);
  const [mensagem, setMensagem] = useState<{texto: string, tipo: 'erro' | 'sucesso'} | null>(null);
  
  // Animations for toast
  const [fadeAnim] = useState(new Animated.Value(0));
  const [slideAnim] = useState(new Animated.Value(-50));

  const mostrarToast = (texto: string, tipo: 'sucesso' | 'erro') => {
    setMensagem({ texto, tipo });
    
    // Reset das animações
    fadeAnim.setValue(0);
    slideAnim.setValue(-50);
    
    // Animação de entrada
    Animated.parallel([
      Animated.timing(fadeAnim, {
        toValue: 1,
        duration: 300,
        useNativeDriver: true,
      }),
      Animated.timing(slideAnim, {
        toValue: 0,
        duration: 300,
        useNativeDriver: true,
      }),
    ]).start();

    // Auto-hide após 3 segundos
    setTimeout(() => {
      esconderToast();
    }, 3000);
  };

  const esconderToast = () => {
    Animated.parallel([
      Animated.timing(fadeAnim, {
        toValue: 0,
        duration: 300,
        useNativeDriver: true,
      }),
      Animated.timing(slideAnim, {
        toValue: -50,
        duration: 300,
        useNativeDriver: true,
      }),
    ]).start(() => {
      setMensagem(null);
    });
  };

  const handleSimulacao = async () => {
    if (!valor || !prazo || !produtoSelecionado) {
      mostrarToast('Preencha o valor, o prazo e selecione um produto.', 'erro');
      return;
    }
    setIsSimulating(true);

    const valorFloat = parseFloat(valor);
    const prazoInt = parseInt(prazo);

    if (prazoInt <= 0 || prazoInt > produtoSelecionado.prazoMaximo) {
      mostrarToast(
        `O prazo para este produto deve ser entre 1 e ${produtoSelecionado.prazoMaximo} meses.`, 
        'erro'
      );
      setIsSimulating(false);
      return;
    }

    try {
      let resultado;
      
      if (postSimulacao) {
        // Usa API real para simulação
        resultado = await postSimulacao({
          produtoId: produtoSelecionado.id,
          valor: valorFloat,
          prazo: prazoInt,
        });
      } else {
        // Usa cálculo local (modo mock)
        await new Promise(resolve => setTimeout(resolve, 1000));
        resultado = calcularSimulacao(valorFloat, prazoInt, produtoSelecionado);
      }
      
      setResultado(resultado);
      setModalVisible(true);

    } catch (error) {
      console.error("Erro ao simular:", error);
      
      if (error instanceof Error) {
        mostrarToast(error.message, 'erro');
      } else {
        mostrarToast('Não foi possível conectar ao servidor para realizar a simulação.', 'erro');
      }
    } finally {
      setIsSimulating(false);
    }
  };

  const handleOpenModal = () => {
    if (process.env.JEST_WORKER_ID) {
      setModalProdutoVisible(true); // Directly set modal visibility in test environment
      return;
    }
    setModalProdutoVisible(true);
  };

  return (
    <React.Fragment>
      <ScrollView 
        style={{ 
          flex: 1, 
          backgroundColor: '#FFFFFF' 
        }} 
        contentContainerStyle={{
          flexGrow: 1, 
          padding: 20, 
          justifyContent: 'flex-start', 
          alignItems: 'center', 
          maxWidth: 600, 
          width: '100%', 
          alignSelf: 'center',
          backgroundColor: '#FFFFFF',
          paddingBottom: 40
        }}
        showsVerticalScrollIndicator={false}
        keyboardShouldPersistTaps="handled"
        bounces={false}
      >
        <Text style={globalStyles.title}>Simular Empréstimo</Text>
        
        {/* Toast Notification */}
        {mensagem && (
          <Animated.View 
            style={[
              toastStyles.container,
              mensagem.tipo === 'erro' ? toastStyles.erro : toastStyles.sucesso,
              {
                opacity: fadeAnim,
                transform: [{ translateY: slideAnim }]
              }
            ]}
          >
            <Text style={toastStyles.texto}>{mensagem.texto}</Text>
          </Animated.View>
        )}
        
        <Text style={globalStyles.label}>Selecione o Produto:</Text>
        
        <ProdutoCompactSelector
          produtos={produtos}
          produtoSelecionado={produtoSelecionado}
          onOpenModal={handleOpenModal}
          onSelectProduto={setProdutoSelecionado}
        />
        
        <TextInput 
          style={globalStyles.input} 
          placeholder="Valor do Empréstimo (R$)" 
          placeholderTextColor="#888" 
          keyboardType="numeric" 
          value={valor} 
          onChangeText={setValor}
        />
        <TextInput 
          style={globalStyles.input} 
          placeholder="Prazo (meses)" 
          placeholderTextColor="#888" 
          keyboardType="numeric" 
          value={prazo} 
          onChangeText={setPrazo}
        />
        <TouchableOpacity 
          style={[globalStyles.buttonOrange, isSimulating && globalStyles.buttonDisabled]} 
          onPress={handleSimulacao} 
          disabled={isSimulating}
          testID="open-modal-button"
        >
          {isSimulating ? 
            <ActivityIndicator color="#fff" /> : 
            <Text style={globalStyles.buttonText}>Simular</Text>
          }
        </TouchableOpacity>
        <TouchableOpacity 
          style={[globalStyles.button, globalStyles.buttonSecondary]} 
          onPress={goHome}
        >
          <Text style={globalStyles.buttonTextSecondary}>Voltar</Text>
        </TouchableOpacity>
      </ScrollView>

      {modalProdutoVisible && (
        <ProdutoSelectorModal
          visible={modalProdutoVisible}
          produtos={produtos}
          produtoSelecionado={produtoSelecionado}
          onSelectProduto={setProdutoSelecionado}
          onClose={() => setModalProdutoVisible(false)}
        />
      )}
    </React.Fragment>
  );
};

const toastStyles = StyleSheet.create({
  container: {
    position: 'absolute',
    top: 20,
    left: 20,
    right: 20,
    padding: 16,
    borderRadius: 8,
    zIndex: 9999,
    elevation: 999,
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.25,
    shadowRadius: 4,
  },
  sucesso: {
    backgroundColor: '#4CAF50',
  },
  erro: {
    backgroundColor: '#F44336',
  },
  texto: {
    color: '#FFFFFF',
    fontSize: 16,
    fontWeight: 'bold',
    textAlign: 'center',
    fontFamily: 'Caixa',
  },
});

export default SimulacaoScreen;
