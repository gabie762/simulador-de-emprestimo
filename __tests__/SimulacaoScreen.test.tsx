import { fireEvent, render, waitFor } from '@testing-library/react-native';
import React from 'react';
import SimulacaoScreen from '../screens/SimulacaoScreen';
import { postSimulacao } from '../utils/apiConfig'; // Corrected import path

jest.mock('../utils/apiConfig', () => ({
  postSimulacao: jest.fn(),
})); // Properly mock the `postSimulacao` function

jest.useFakeTimers(); // Apply fake timers globally

beforeEach(() => {
  jest.clearAllMocks();
});

afterEach(() => {
  jest.clearAllTimers(); // Clear all timers after each test
});

// Suppress console errors during tests
beforeAll(() => {
  jest.spyOn(console, 'error').mockImplementation(() => {});
});

afterAll(() => {
  jest.restoreAllMocks();
});

describe('SimulacaoScreen', () => {
  const mockGoHome = jest.fn();
  const mockSetResultado = jest.fn();
  const mockSetModalVisible = jest.fn();
  const mockProdutos = [
    { id: '1', nome: 'Produto 1', prazoMaximo: 12, taxaJurosAnual: 5.0 },
    { id: '2', nome: 'Produto 2', prazoMaximo: 24, taxaJurosAnual: 6.5 },
  ]; // Added missing property `taxaJurosAnual`

  it('renders correctly', () => {
    const { toJSON } = render(
      <SimulacaoScreen
        produtos={[]}
        goHome={mockGoHome}
        setResultado={mockSetResultado}
        setModalVisible={mockSetModalVisible}
      />
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('shows an error toast for missing inputs', async () => {
    const { getByText } = render(
      <SimulacaoScreen
        produtos={mockProdutos}
        goHome={mockGoHome}
        setResultado={mockSetResultado}
        setModalVisible={mockSetModalVisible}
      />
    );

    fireEvent.press(getByText('Simular'));

    await waitFor(() => {
      expect(getByText('Preencha o valor, o prazo e selecione um produto.')).toBeTruthy();
    });
  });

  it('shows an error toast for invalid prazo', async () => {
    const { getByText, getByPlaceholderText } = render(
      <SimulacaoScreen
        produtos={mockProdutos}
        goHome={mockGoHome}
        setResultado={mockSetResultado}
        setModalVisible={mockSetModalVisible}
      />
    );

    fireEvent.changeText(getByPlaceholderText('Valor do Empréstimo (R$)'), '1000');
    fireEvent.changeText(getByPlaceholderText('Prazo (meses)'), '25'); // Invalid prazo
    fireEvent.press(getByText('Simular'));

    await waitFor(() => {
      expect(getByText('O prazo para este produto deve ser entre 1 e 12 meses.')).toBeTruthy();
    });
  });

  it('calls postSimulacao for valid inputs', async () => {
    (postSimulacao as jest.Mock).mockResolvedValueOnce({ resultado: 'Simulação bem-sucedida' });

    const { getByText, getByPlaceholderText } = render(
      <SimulacaoScreen
        produtos={mockProdutos}
        goHome={mockGoHome}
        setResultado={mockSetResultado}
        setModalVisible={mockSetModalVisible}
      />
    );

    fireEvent.changeText(getByPlaceholderText('Valor do Empréstimo (R$)'), '1000');
    fireEvent.changeText(getByPlaceholderText('Prazo (meses)'), '12');
    fireEvent.press(getByText('Simular'));

    await waitFor(() => {
      expect(postSimulacao).toHaveBeenCalledWith({
        produtoId: '1',
        valor: 1000,
        prazo: 12,
      });
      expect(mockSetResultado).toHaveBeenCalledWith({ resultado: 'Simulação bem-sucedida' });
      expect(mockSetModalVisible).toHaveBeenCalledWith(true);
    });
  });

  it('shows an error toast if postSimulacao fails', async () => {
    (postSimulacao as jest.Mock).mockRejectedValueOnce(new Error('Erro na simulação'));

    const { getByText, getByPlaceholderText } = render(
      <SimulacaoScreen
        produtos={mockProdutos}
        goHome={mockGoHome}
        setResultado={mockSetResultado}
        setModalVisible={mockSetModalVisible}
      />
    );

    fireEvent.changeText(getByPlaceholderText('Valor do Empréstimo (R$)'), '1000');
    fireEvent.changeText(getByPlaceholderText('Prazo (meses)'), '12');
    fireEvent.press(getByText('Simular'));

    await waitFor(() => {
      expect(getByText('Erro na simulação')).toBeTruthy();
    });
  });

  it('opens and closes the ProdutoSelectorModal', () => {
    const { getByTestId, queryByText } = render(
      <SimulacaoScreen
        produtos={mockProdutos}
        goHome={mockGoHome}
        setResultado={mockSetResultado}
        setModalVisible={mockSetModalVisible}
      />
    );

    // Modal should not be visible initially
    expect(queryByText('Selecionar Produto')).toBeNull();

    // Open modal
    fireEvent.press(getByTestId('produto-selecionado-card'));

    // Modal should now be visible
    expect(queryByText('Selecionar Produto')).toBeTruthy();
  });

  it('navigates back when Voltar is pressed', () => {
    const { getByText } = render(
      <SimulacaoScreen
        produtos={mockProdutos}
        goHome={mockGoHome}
        setResultado={mockSetResultado}
        setModalVisible={mockSetModalVisible}
      />
    );

    fireEvent.press(getByText('Voltar'));
    expect(mockGoHome).toHaveBeenCalled();
  });

  it('handles toast auto-hide timeout', async () => {
    const { getByText } = render(
      <SimulacaoScreen
        produtos={mockProdutos}
        goHome={mockGoHome}
        setResultado={mockSetResultado}
        setModalVisible={mockSetModalVisible}
      />
    );

    // Trigger toast by clicking simulate without inputs
    fireEvent.press(getByText('Simular'));
    
    await waitFor(() => {
      expect(getByText('Preencha o valor, o prazo e selecione um produto.')).toBeTruthy();
    });

    // Fast-forward timers to trigger auto-hide
    jest.advanceTimersByTime(3000);
    
    await waitFor(() => {
      expect(() => getByText('Preencha o valor, o prazo e selecione um produto.')).toThrow();
    });
  });

  it('handles zero or negative prazo values', async () => {
    const { getByText, getByPlaceholderText } = render(
      <SimulacaoScreen
        produtos={mockProdutos}
        goHome={mockGoHome}
        setResultado={mockSetResultado}
        setModalVisible={mockSetModalVisible}
      />
    );

    // Test zero prazo
    fireEvent.changeText(getByPlaceholderText('Valor do Empréstimo (R$)'), '1000');
    fireEvent.changeText(getByPlaceholderText('Prazo (meses)'), '0');
    fireEvent.press(getByText('Simular'));

    await waitFor(() => {
      expect(getByText('O prazo para este produto deve ser entre 1 e 12 meses.')).toBeTruthy();
    });
  });

  it('handles local calculation when postSimulacao is null', async () => {
    // Mock postSimulacao as null to trigger local calculation
    const originalMock = jest.requireMock('../utils/apiConfig');
    originalMock.postSimulacao = null;

    const { getByText, getByPlaceholderText } = render(
      <SimulacaoScreen
        produtos={mockProdutos}
        goHome={mockGoHome}
        setResultado={mockSetResultado}
        setModalVisible={mockSetModalVisible}
      />
    );

    fireEvent.changeText(getByPlaceholderText('Valor do Empréstimo (R$)'), '1000');
    fireEvent.changeText(getByPlaceholderText('Prazo (meses)'), '12');
    fireEvent.press(getByText('Simular'));

    // Fast-forward the 1-second delay in local calculation
    jest.advanceTimersByTime(1000);

    await waitFor(() => {
      expect(mockSetResultado).toHaveBeenCalled();
      expect(mockSetModalVisible).toHaveBeenCalledWith(true);
    });

    // Restore original
    originalMock.postSimulacao = jest.fn();
  });
});
