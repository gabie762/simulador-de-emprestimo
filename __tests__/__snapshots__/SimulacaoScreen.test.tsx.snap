// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimulacaoScreen renders correctly 1`] = `
<RCTScrollView
  bounces={false}
  contentContainerStyle={
    {
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": "#FFFFFF",
      "flexGrow": 1,
      "justifyContent": "flex-start",
      "maxWidth": 600,
      "padding": 20,
      "paddingBottom": 40,
      "width": "100%",
    }
  }
  keyboardShouldPersistTaps="handled"
  showsVerticalScrollIndicator={false}
  style={
    {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <View>
    <Text
      style={
        {
          "color": "#0066B3",
          "fontFamily": "Caixa",
          "fontSize": 28,
          "fontWeight": "bold",
          "marginBottom": 20,
          "textAlign": "center",
          "width": "100%",
        }
      }
    >
      Simular Empréstimo
    </Text>
    <Text
      style={
        {
          "alignSelf": "flex-start",
          "color": "#0066B3",
          "fontFamily": "Caixa",
          "fontSize": 16,
          "marginBottom": 10,
        }
      }
    >
      Selecione o Produto:
    </Text>
    <View
      style={
        {
          "marginBottom": 28,
          "paddingHorizontal": 0,
          "paddingVertical": 14,
          "width": "100%",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 16,
            "paddingHorizontal": 2,
            "paddingVertical": 4,
            "width": "100%",
          }
        }
      >
        <View
          accessibilityLabel="Previous"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": true,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "justifyContent": "center",
              "minWidth": 40,
              "opacity": 0.3,
              "paddingHorizontal": 12,
              "paddingVertical": 8,
            }
          }
        >
          <Text
            style={
              {
                "color": "#0066B3",
                "fontSize": 24,
                "fontWeight": "bold",
              }
            }
          >
            ‹
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#F8F9FA",
              "borderColor": "#0066B3",
              "borderRadius": 12,
              "borderWidth": 1,
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginHorizontal": 4,
              "minHeight": 140,
              "opacity": 1,
              "padding": 20,
            }
          }
          testID="produto-selecionado-card"
        >
          <View
            style={
              {
                "flex": 1,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#0066B3",
                  "fontFamily": "Caixa",
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "marginBottom": 10,
                }
              }
            >
              Selecione um produto
            </Text>
          </View>
        </View>
        <View
          accessibilityLabel="Next"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": true,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "justifyContent": "center",
              "minWidth": 40,
              "opacity": 0.3,
              "paddingHorizontal": 12,
              "paddingVertical": 8,
            }
          }
        >
          <Text
            style={
              {
                "color": "#0066B3",
                "fontSize": 24,
                "fontWeight": "bold",
              }
            }
          >
            ›
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 8,
            "justifyContent": "center",
            "marginTop": 12,
          }
        }
      />
    </View>
    <TextInput
      keyboardType="numeric"
      onChangeText={[Function]}
      placeholder="Valor do Empréstimo (R$)"
      placeholderTextColor="#888"
      style={
        {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#C5C7C8",
          "borderRadius": 8,
          "borderWidth": 1,
          "color": "#0066B3",
          "fontFamily": "Caixa",
          "fontSize": 16,
          "marginBottom": 15,
          "paddingHorizontal": 15,
          "paddingVertical": 12,
          "width": "100%",
        }
      }
      value=""
    />
    <TextInput
      keyboardType="numeric"
      onChangeText={[Function]}
      placeholder="Prazo (meses)"
      placeholderTextColor="#888"
      style={
        {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#C5C7C8",
          "borderRadius": 8,
          "borderWidth": 1,
          "color": "#0066B3",
          "fontFamily": "Caixa",
          "fontSize": 16,
          "marginBottom": 15,
          "paddingHorizontal": 15,
          "paddingVertical": 12,
          "width": "100%",
        }
      }
      value=""
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#F7941E",
          "borderRadius": 8,
          "marginBottom": 10,
          "opacity": 1,
          "padding": 15,
          "width": "100%",
        }
      }
      testID="open-modal-button"
    >
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontFamily": "Caixa",
            "fontSize": 18,
            "fontWeight": "bold",
          }
        }
      >
        Simular
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderColor": "#0066B3",
          "borderRadius": 8,
          "borderWidth": 1,
          "marginBottom": 10,
          "opacity": 1,
          "padding": 15,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          {
            "color": "#0066B3",
            "fontFamily": "Caixa",
            "fontSize": 18,
            "fontWeight": "bold",
          }
        }
      >
        Voltar
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
